From 091fd973065ee2aa5030058bde9f9cbf7fec4c41 Mon Sep 17 00:00:00 2001
From: Manjunatha Venkatesh <manjunatha.venkatesh@nxp.com>
Date: Tue, 4 Apr 2017 12:35:13 +0530
Subject: [PATCH 09/10] [artf241278][SPI Module] : SPI Open failing with wrong
 PID execption

During Transcieve in between,ESE access state is changed to P61_STATE_JCP_DWNLD,once it got updated
to this state the subsequent of the new session is blocked because the driver in in
JCOP download state,when ESE open its get failed with PID execption.
---
 drivers/nxp/pn8xT/pn553-i2c/pn553.h             |  2 +-
 external/libese-spi/p73/inc/phNxpEse_Api.h      |  8 --------
 external/libese-spi/p73/lib/phNxpEse_Api.c      |  7 -------
 external/libese-spi/p73/lib/phNxpEse_Internal.h | 15 +++++++++++++++
 4 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/drivers/nxp/pn8xT/pn553-i2c/pn553.h b/drivers/nxp/pn8xT/pn553-i2c/pn553.h
index 582895b..1fb3c5b 100644
--- a/drivers/nxp/pn8xT/pn553-i2c/pn553.h
+++ b/drivers/nxp/pn8xT/pn553-i2c/pn553.h
@@ -117,7 +117,7 @@ typedef enum chip_type_pwr_scheme{
 }chip_pwr_scheme_t;
 
 typedef enum jcop_dwnld_state{
-    JCP_DWNLD_IDLE,                         /* jcop dwnld is not ongoing*/
+    JCP_DWNLD_IDLE = P61_STATE_JCP_DWNLD,   /* jcop dwnld is ongoing*/
     JCP_DWNLD_INIT,                         /* jcop dwonload init state*/
     JCP_DWNLD_START,                        /* download started */
     JCP_SPI_DWNLD_COMPLETE,                 /* jcop download complete in spi interface*/
diff --git a/external/libese-spi/p73/inc/phNxpEse_Api.h b/external/libese-spi/p73/inc/phNxpEse_Api.h
index e388b60..4385bef 100644
--- a/external/libese-spi/p73/inc/phNxpEse_Api.h
+++ b/external/libese-spi/p73/inc/phNxpEse_Api.h
@@ -36,14 +36,6 @@ typedef struct phNxpEse_data
     uint8_t  *p_data; /*!< pointer to a buffer */
 } phNxpEse_data;
 
-typedef enum jcop_dwnld_state{
-    JCP_DWNLD_IDLE,                         /* jcop dwnld is not ongoing*/
-    JCP_DWNLD_INIT,                         /* jcop dwonload init state*/
-    JCP_DWNLD_START,                        /* download started */
-    JCP_SPI_DWNLD_COMPLETE,                 /* jcop download complete in spi interface*/
-    JCP_DWP_DWNLD_COMPLETE,                 /* jcop download complete */
-} phNxpEse_JcopDwnldState;
-
 /**
  * \ingroup spi_libese
  * \brief Ese Channel mode
diff --git a/external/libese-spi/p73/lib/phNxpEse_Api.c b/external/libese-spi/p73/lib/phNxpEse_Api.c
index fd13021..07bd94b 100755
--- a/external/libese-spi/p73/lib/phNxpEse_Api.c
+++ b/external/libese-spi/p73/lib/phNxpEse_Api.c
@@ -22,13 +22,6 @@
 #include <NXP_ESE_FEATURES.h>
 #include <phNxpEsePal_spi.h>
 
-/* Macro to enable SPM Module */
-#define SPM_INTEGRATED
-//#undef SPM_INTEGRATED
-#ifdef SPM_INTEGRATED
-#include "../spm/phNxpEse_Spm.h"
-#endif
-
 #define RECIEVE_PACKET_SOF      0xA5
 static int phNxpEse_readPacket(void *pDevHandle, uint8_t * pBuffer, int nNbBytesToRead);
 static ESESTATUS phNxpEse_checkJcopDwnldState(void);
diff --git a/external/libese-spi/p73/lib/phNxpEse_Internal.h b/external/libese-spi/p73/lib/phNxpEse_Internal.h
index 5bcc886..1fdc2a9 100644
--- a/external/libese-spi/p73/lib/phNxpEse_Internal.h
+++ b/external/libese-spi/p73/lib/phNxpEse_Internal.h
@@ -19,6 +19,12 @@
 #include <phNxpEse_Api.h>
 #include <phNxpLog.h>
 
+/* Macro to enable SPM Module */
+#define SPM_INTEGRATED
+//#undef SPM_INTEGRATED
+#ifdef SPM_INTEGRATED
+#include "../spm/phNxpEse_Spm.h"
+#endif
 /********************* Definitions and structures *****************************/
 
 typedef enum
@@ -46,6 +52,15 @@ typedef enum
 #define ESE_FW_DWNLD_RETRY_CNT        10 /* Maximum retry count for FW Dwonload*/
 #endif
 
+/* JCOP download states */
+typedef enum jcop_dwnld_state{
+    JCP_DWNLD_IDLE = SPM_STATE_JCOP_DWNLD,  /* jcop dwnld is not ongoing*/
+    JCP_DWNLD_INIT,                         /* jcop dwonload init state*/
+    JCP_DWNLD_START,                        /* download started */
+    JCP_SPI_DWNLD_COMPLETE,                 /* jcop download complete in spi interface*/
+    JCP_DWP_DWNLD_COMPLETE,                 /* jcop download complete */
+} phNxpEse_JcopDwnldState;
+
 /* SPI Control structure */
 typedef struct phNxpEse_Context
 {
-- 
1.9.1

